storage/tokudb/ha_tokudb.cc

```diff
@@ -152,17 +152,15 @@ void TOKUDB_SHARE::hash_free_element(TOKUDB_SHARE* share) {
     delete share;
 }
 void TOKUDB_SHARE::static_init() {
-    my_hash_init(
-        &_open_tables,
-        table_alias_charset,
-        32,
-        0,
-        0,
-        (my_hash_get_key)hash_get_key,
-        (my_hash_free_key)hash_free_element, 0);
+    assert_always(_open_tables.size() == 0);
 }
 void TOKUDB_SHARE::static_destroy() {
-    my_hash_free(&_open_tables);
+    for (auto it = _open_tables.cbegin(); it != _open_tables.cend(); ) {
+        TOKUDB_TRACE("_open_tables %s %p", it->first.c_str(), it->second);
+        hash_free_element(it->second);
+        it = _open_tables.erase(it);
```

Comment:
Can you just iterate over the contents here and do clear() after the loop?

Reply:
True, and actually, this shows another minor problem, there is no need for hash_free_element, the code should just do
`it->second->destroy();
delete it->second;`